### PR TITLE
feat(tracing): Add line information from tracing

### PR DESCRIPTION
**Pull request by flub**
This adds the line information that tracing provides back into the
event.  It also moves around the tags into a custom context instead of
the "extra" grab-bag as this is less likely to step on the user's
ad-hoc toes.

Related to #427

### CODE REVIEW

**Review by loewenheim — APPROVED** [review]

File: sentry-tracing/src/converters.rs [hunk 1]
```diff
@@ -103,6 +103,36 @@ pub fn breadcrumb_from_event(event: &tracing_core::Event) -> Breadcrumb {
...
+        location_map.insert("file".to_string(), file.into());
+    }
+    if let Some(line) = event_meta.line() {
+        location_map.insert("line".to_string(), line.into());
+    }
+
+    let mut context = BTreeMap::new();
+    context.insert(
+        "Rust Tracing Tags".to_string(),
+        sentry_core::protocol::Context::Other(event_tags),
```
Comment (Swatinem) [inline sentry-tracing/src/converters.rs:125]: can we also wrap all of this in a `event_tags.is_empty` check?

File: sentry-tracing/src/converters.rs [hunk 2]
```diff
@@ -103,6 +103,36 @@ pub fn breadcrumb_from_event(event: &tracing_core::Event) -> Breadcrumb {
...
+    );
+    if !location_map.is_empty() {
+        context.insert(
+            "Rust Tracing Location".to_string(),
```
Comment (Swatinem) [inline sentry-tracing/src/converters.rs:129]: its quite unfortunate we don’t have any other event attrs for source location :-(

File: sentry/tests/test_tracing.rs [hunk 3]
```diff
@@ -64,6 +80,21 @@ fn test_tracing() {
...
+            let value = Value::String("tagvalue".to_string());
+            assert_eq!(*tags.get("tagname").unwrap(), value);
+        }
+        _ => panic!("Wrong context type"),
+    }
+    match event.contexts.get("Rust Tracing Location").unwrap() {
+        Context::Other(tags) => {
+            assert!(matches!(tags.get("module_path").unwrap(), Value::String(_)));
+            assert!(matches!(tags.get("file").unwrap(), Value::String(_)));
+            assert!(matches!(tags.get("line").unwrap(), Value::Number(_)));
```
Comment (Swatinem) [inline sentry/tests/test_tracing.rs:94]: might as well assert the file/line ;-)

**Review by Swatinem — APPROVED** [review]

**Review by flub — COMMENTED** [review]
There are like lots more weird edge cases that i just don't understand what tracing is doing - which is why there's no test for the empty tags.  I'm giving up though as this seems hopeless, I think the current implementation gives some useful information so I'm going to close my eyes and click merge.
